ruff_dev/src/generate_rules_table.rs

```diff
@@ -23,77 +26,90 @@ pub struct Cli {
+            check_category.title().to_lowercase().replace(" ", "-"),
+            codes_csv.to_lowercase().replace(",", "-").replace(" ", "")
         ));
-        output.push('\n');
-        output.push('\n');
 
         if let Some((url, platform)) = check_category.url() {
-            output.push_str(&format!(
+            table_out.push_str(&format!(
                 "For more, see [{}]({}) on {}.",
                 check_category.title(),
                 url,
                 platform
             ));
-            output.push('\n');
-            output.push('\n');
+            table_out.push('\n');
+            table_out.push('\n');
         }
 
-        output.push_str("| Code | Name | Message | Fix |");
-        output.push('\n');
-        output.push_str("| ---- | ---- | ------- | --- |");
-        output.push('\n');
+        table_out.push_str("| Code | Name | Message | Fix |");
+        table_out.push('\n');
+        table_out.push_str("| ---- | ---- | ------- | --- |");
+        table_out.push('\n');
 
         for check_code in CheckCode::iter() {
             if check_code.category() == check_category {
                 let check_kind = check_code.kind();
                 let fix_token = if check_kind.fixable() { "🛠" } else { "" };
-                output.push_str(&format!(
+                table_out.push_str(&format!(
                     "| {} | {} | {} | {} |",
                     check_kind.code().as_ref(),
                     check_kind.as_ref(),
                     check_kind.summary().replace('|', r"\|"),
                     fix_token
                 ));
-                output.push('\n');
+                table_out.push('\n');
             }
         }
-        output.push('\n');
+        table_out.push('\n');
     }
 
     if cli.dry_run {
-        print!("{output}");
+        print!("Table of Contents: {toc_out}");
     } else {
-        // Read the existing file.
-        let file = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
-            .parent()
-            .expect("Failed to find root directory")
-            .join("README.md");
-        let existing = fs::read_to_string(&file)?;
-
-        // Extract the prefix.
-        let index = existing
-            .find(BEGIN_PRAGMA)
-            .expect("Unable to find begin pragma");
-        let prefix = &existing[..index + BEGIN_PRAGMA.len()];
-
-        // Extract the suffix.
-        let index = existing
-            .find(END_PRAGMA)
-            .expect("Unable to find end pragma");
-        let suffix = &existing[index..];
-
-        // Write the prefix, new contents, and suffix.
-        let mut f = OpenOptions::new().write(true).truncate(true).open(&file)?;
-        write!(f, "{prefix}\n\n")?;
-        write!(f, "{output}")?;
-        write!(f, "{suffix}")?;
+        let _toc_ok = replace_readme_section(toc_out, TOC_BEGIN_PRAGMA, TOC_END_PRAGMA);
+        let _table_ok = replace_readme_section(table_out, TABLE_BEGIN_PRAGMA, TABLE_END_PRAGMA);
```

Comment:
I wasn't sure what the "right" way to handle multiple results here. I wanted to do something like:

```rust
return Result(toc_ok, table_ok)
```

or pythonic-like:

```python
return toc_ok and table_ok
```

Reply:
See my other comment!

If you're calling a function that returns `Result<T>` from a function that returns `Result<T>` (as is the case here), you can always do `let ok_value = foo()?`, which will leave `ok_value` as type `T` but return immediately if `foo` returned an error.